### ISSUE TITLE
Collision distance for mesh models does not return correct results when objects are in collision

### ISSUE THREAD

**Issue by SLrepo** [flexible-collision-library/fcl#491]
According to the reference, collision distance should be negative when objects are in collision. 
However, when the objects are not primitive shapes (e.g. when a loaded .obj file object is in collision with some other objects), the min_distance returned by DistanceResult is always 0. 
Below is an example. 

>    BVHModel<RSS<double>> m1; 
  std::vector<Vector3<double>> p1;
  std::vector<Triangle> t1;
  loadOBJFile("mesh.obj", p1, t1);
  m1.beginModel();
  m1.addSubModel(p1, t1);
  m1.endModel();
  BVHModel<RSS<double>> m2;
  Cylinder<double> c2(20, 40);
  generateBVHModel(m2, c2, Transform3<double>::Identity(), 16, 16);
  Transform3<double> tf1 = Transform3<double>::Identity();
  tf1.translation() = Vector3<double>(20, 0, 0);
  DistanceRequest<double> request;
  request.enable_signed_distance = true;
  DistanceResult<double> res;
  res.clear();
  distance(&m1, Transform3<double>::Identity(), &m2, tf1, request, res);
  std::cout << "The penetration depth is " << res.min_distance << "\n";

When I change the first axis of "tf1" from 0 to 20, the returned min_distance gradually become smaller, at some point, it should become negative (objects in collision). However, it never becomes negative and stayed at 0. The .obj file I used is from [here](https://free3d.com/3d-model/anime-character-high-53650.html), the 'loadOBJFile' function is the same as the one used in fcl/test.

**Comment by jmirabel** [comment]
FCL cannot compute the penetration between a non-convex mesh and anything else, for several reasons.
1. BVHModel may not even be closed so the interior or exterior is not well defined. Thus BVHModel are surface only.
2. Even if you define conventionally an interior, for a non-convex model, computing the penetration depth is not a local problem any more and is thus hard (read time-consuming) to compute. This isn't implemented in FCL.

**Comment by SLrepo** [comment]
> FCL cannot compute the penetration between a non-convex mesh and anything else, for several reasons.
> 
> 1. BVHModel may not even be closed so the interior or exterior is not well defined. Thus BVHModel are surface only.
> 2. Even if you define conventionally an interior, for a non-convex model, computing the penetration depth is not a local problem any more and is thus hard (read time-consuming) to compute. This isn't implemented in FCL.

Does it mean if the meshes are convex, fcl would work? Since a non-convex shape can be decomposed into several convex shapes.

**Comment by SeanCurtis-TRI** [comment]
Correct. The key being that convex shapes, by definition, *must* be closed.  :)  There is no ambiguity about inside/outside and there will be a well-defined minimum-translational distance (even though the "nearest point" may not be unique.)
